stage/src/tasks/agg_task.rs

```diff
@@ -37,3 +39,157 @@ impl Clone for AggAllTask {
         }
     }
 }
+
+#[derive(Debug, Default, Deserialize, Serialize)]
+pub struct AggTask {
+    pub file_key: String,
+    pub task_id: String,
+    pub state: u32,
+    pub base_dir: String,
+    pub block_no: u64,
+    pub seg_size: u32,
+    pub proof_id: String,
+    pub proof_path1: String,
+    pub proof_path2: String,
+    pub pub_value_path1: String,
+    pub pub_value_path2: String,
+    pub is_agg1: bool,
+    pub is_agg2: bool,
+    pub is_final: bool,
+    pub from_prove: bool,
+    pub output_proof_path: String,
+    pub output_pub_value_path: String,
+    pub output_dir: String,
+    pub start_ts: u64,
+    pub finish_ts: u64,
+    pub node_info: String,
+
+    // depend
+    pub left: Option<String>,
+    pub right: Option<String>,
+}
+
+impl Clone for AggTask {
+    fn clone(&self) -> Self {
+        AggTask {
```

Comment:
Can we create another structure, like RootProof, which consists of proof_path, pub_value_path,  is_agg. Then,  we can define 2 fields in AggTask,  left: Option<RootProof>, right: Option<RootProof>.

Reply:
done